tests/path.py

```diff
@@ -205,3 +207,18 @@ def listdir(self):
 
     def __repr__(self):
         return '%s(%s)' % (self.__class__.__name__, text_type.__repr__(self))
+
+
+# Lives here only to avoid circular references;  use it from util.py!
```

Comment:
Note these types (`_repr_text` and `_repr_bin`) do *NOT* survive any string operations on them (ie., `type(_repr_text("foo") + "bar") == text_type`.)  I'd argue that for the purposes at hand that's good enough.